code/datums/elements/art.dm

```diff
@@ -0,0 +1,70 @@
+/datum/element/art
+	element_flags = ELEMENT_BESPOKE|ELEMENT_DETACH
+	id_arg_index = 2
+	var/impressiveness = 0
+
+/datum/element/art/Attach(datum/target, impress)
+	. = ..()
+	if(!isatom(target) || isarea(target))
+		return ELEMENT_INCOMPATIBLE
+	impressiveness = impress
+	if(isobj(target))
+		RegisterSignal(target, COMSIG_PARENT_EXAMINE, .proc/on_obj_examine)
+		if(isstructure(target))
+			RegisterSignal(target, COMSIG_ATOM_ATTACK_HAND, .proc/on_attack_hand)
+		if(isitem(target))
+			RegisterSignal(target, COMSIG_ITEM_ATTACK_SELF, .proc/apply_moodlet)
```

Comment:
This will end up in `apply_moodlet` being called with a `null` impressiveness value.
This should be handled by an intermediate proc, something like `.proc/on_item_attack_self`, which calls `apply_moodlet`